executor/show.go

```diff
@@ -402,11 +402,33 @@ func (e *ShowExec) fetchShowDatabases() error {
 	dbs := e.is.AllSchemaNames()
 	checker := privilege.GetPrivilegeManager(e.ctx)
 	sort.Strings(dbs)
+	var (
+		lowerCaseNames    = make(map[string]string)
+		fieldPatternsLike collate.WildcardPattern
+		FieldFilterEnable bool
+		fieldFilter       string
+	)
+	for _, info := range e.is.AllSchemas() {
+		lowerCaseNames[info.Name.O] = info.Name.L
+	}
+	if e.Extractor != nil {
+		extractor := (e.Extractor).(*plannercore.ShowDatabaseExtractor)
+		if extractor.FieldPatterns != "" {
+			fieldPatternsLike = collate.GetCollatorByID(collate.CollationName2ID(mysql.UTF8MB4DefaultCollation)).Pattern()
+			fieldPatternsLike.Compile(extractor.FieldPatterns, byte('\\'))
+		}
+		FieldFilterEnable = extractor.Field != ""
+		fieldFilter = extractor.Field
+	}
 	// let information_schema be the first database
 	moveInfoSchemaToFront(dbs)
 	for _, d := range dbs {
 		if checker != nil && !checker.DBIsVisible(e.ctx.GetSessionVars().ActiveRoles, d) {
 			continue
+		} else if FieldFilterEnable && lowerCaseNames[d] != fieldFilter {
```

Comment:
why not use `strings.ToLower()` to replace the map

Reply:
fixed